### PR TITLE
Added OAuth1 generic connetion and XING sample

### PR DESCRIPTION
**Pull request by eugeniop**
Added OAuth1 generic connetion and XING sample

### CODE REVIEW

File: docs/oauth1.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+To create an arbitrary __OAuth1__ connection, you use __[Auth0's Connections API](api#!#post--api-connections)__. 
+
+This example would create a custom Twitter connection:
+
+```
+curl -H "Content-Type: application/json" -H 'Authorization: Bearer {YOUR_GLOBAL_CLIENT_ACCESS_TOKEN}' -d @twitter.json https://@@account.namespace@@/api/connections
+```
+
+```
```
Comment (dschenkelman) [inline docs/oauth1.md]: Add a note that this is twitter.json?

File: docs/oauth1.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      "fetchUserProfile": "function (token, tokenSecret, ctx, cb) {var OAuth = new require('oauth').OAuth;var oauth = new OAuth(ctx.requestTokenURL,ctx.accessTokenURL,ctx.client_id,ctx.client_secret,'1.0',null,'HMAC-SHA1');oauth.get('https://api.twitter.com/1.1/users/show.json?user_id=' + ctx.user_id,token,tokenSecret,function(e, b, r) {if (e) return cb(e);if (r.statusCode !== 200) return cb(new Error('StatusCode: ' + r.statusCode));cb(null, JSON.parse(b));});}"
+    }
+  }
+}
+```
+
+The key parameters for a connection are:
+
+* **name**: this is how the connection can be referenced later on in Auth0 or in your app.
+* **strategy**: this must be __oauth1__. It defines the protocol implemented by the provider.
```
Comment (dschenkelman) [inline docs/oauth1.md]: use quotes for oauth1?

File: docs/oauth1.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+The __options__ object:
+
+* **client_id** and **client_secret** must be obtained from your provider.
+* **requestTokenURL**: `https://api.twitter.com/oauth/request_token`
+* **accessTokenURL**: `https://api.twitter.com/oauth/access_token`
+* **userAuthorizationURL**: `https://api.twitter.com/oauth/authenticate`
+* **fetchUserProfile**: Auth0 allows you to define a custom script that returns a JSON object with user info. What you do in the script is up to you. For convenience, the __[OAuth](https://www.npmjs.com/package/oauth)__ module is included to simplify OAuth1 calls.
+
+The script will have the following signature:
```
Comment (dschenkelman) [inline docs/oauth1.md]: That's not the signature, that's an example.

Comment (dschenkelman) [inline docs/oauth1.md]: This seems like advanced usage. Should it be in a separate section below ("customize user info") or sthg like that.

**Comment by eugeniop** [comment]
Yes, this is for us for the time being. We might need some reorg as you suggest.But I had someone in my backlog waiting for this, so wanted to get it out of the way.

**Pull request merged**